src/linker/pipeline.py

```diff
@@ -26,18 +24,30 @@ def run(self, runner: Callable, results_dir: Path) -> None:
                 results_dir=results_dir,
             )
 
-    def _validate(self, schema: "PipelineSchema") -> None:
-        for idx, implementation in enumerate(self.implementations):
-            # Check that all steps are accounted for and in the correct order
-            if implementation.step_name != schema.steps[idx].name:
-                return False
-            # Check that container exists
-            if (
-                not Path(f"{implementation._container_full_stem}.tar.gz").exists()
-                and not Path(f"{implementation._container_full_stem}.sif").exists()
-            ):
-                return False
-        return True
+    def _validate(self) -> None:
+        """Validates the pipeline against supported schemas."""
+
+        from linker.pipeline_schema import PIPELINE_SCHEMAS, PipelineSchema
+
+        def validate_pipeline(schema: PipelineSchema):
```

Comment:
I thought this would be more like:

```python
def validate_pipeline(schema: PipelineSchema):
    for idx, implementation in enumerate(self.implementations):
        # Check that all steps are accounted for and in the correct order
        if implementation.step_name != schema.steps[idx].name:
            return False

        implementation.validate()
```

Reply:
I did it slightly differently - see the latest commit. implementation.validate happens when Implementation is instantiated

Reply:
That's what we actually discussed, so looks good